src/System.Linq/src/System/Linq/Select.cs

```diff
@@ -247,9 +298,56 @@ public int GetCount(bool onlyIfCheap)
             {
                 return _source.Count;
             }
+
+            public IPartition<TResult> Skip(int count)
+            {
+                return count == 0 ? (IPartition<TResult>)Clone() : new SelectIPartitionIterator<TSource, TResult>(new ListPartition<TSource>(_source, count, int.MaxValue), _selector);
+            }
+
+            public IPartition<TResult> Take(int count)
+            {
+                return new SelectIPartitionIterator<TSource, TResult>(new ListPartition<TSource>(_source, 0, count - 1), _selector);
+            }
```

Comment:
Same allocation question.  This makes me question the value implementing IPartition here.